hastory-server/src/Data/Hastory/Server/TestUtils.hs

```diff
@@ -56,7 +77,44 @@ withTestServer func = do
           void $ runSqlPool (runMigrationSilent migrateAll) siPool
           let mkApp = pure $ app opts settings
               opts = Options 10 Nothing
-              settings = ServerSettings siToken siPool
+              settings = ServerSettings siPool jwtSettings defaultCookieSettings
           testWithApplication mkApp $ \p ->
             let siClientEnv = mkClientEnv manager (BaseUrl Http "127.0.0.1" p "")
              in func (ServerInfo {..})
+
+type RegistrationData = (UserId, Token)
+
+withNewUser :: ClientEnv -> (RegistrationData -> Expectation) -> Expectation
+withNewUser clientEnv func = do
+  userForm <- randomUserForm
+  Right userId <- runClientM (createUserClient userForm) clientEnv
+  Right resp <- runClientM (createSessionClient userForm) clientEnv
+  case extractJWTCookie resp of
+    Nothing -> expectationFailure "JWT Cookie not found"
+    Just cookie -> func (userId, Token cookie)
+
+extractJWTCookie :: Headers AuthCookies NoContent -> Maybe B.ByteString
+extractJWTCookie headerList = do
+  (_, cookie) <- find jwtCookie (getHeaders headerList)
+  pure . B.takeWhile (/= ';') $ B.drop 11 cookie
+  where
+    jwtCookie (headerName, headerValue) =
+      headerName == "Set-Cookie" && B.take 11 headerValue == "JWT-Cookie="
+
+createEntry :: ClientEnv -> Token -> SyncRequest -> IO (Either ClientError NoContent)
+createEntry clientEnv token syncReq = runClientM (createEntryClient token syncReq) clientEnv
+
+createUser :: ClientEnv -> UserForm -> IO (Either ClientError UserId)
+createUser clientEnv userForm = runClientM (createUserClient userForm) clientEnv
+
+getEntries :: Pool SqlBackend -> IO [Entity ServerEntry]
+getEntries = runSqlPool (selectList [] [])
+
+getUsers :: Pool SqlBackend -> IO [Entity User]
+getUsers = runSqlPool (selectList [] [])
```

Comment:
I don't think it's worth making functions for these.
In particular because `getEntries` is probably not what you want because you'd be mixing users' entries.
Instead, you can try using a custom monad for the server handler: https://github.com/NorfairKing/intray/blob/f30c5deff1666a0601a64a1a9c696714f0558e6a/intray-server/src/Intray/Server/Types.hs#L33
and then write a `runDB` (this is idiomatic): 
https://github.com/NorfairKing/intray/blob/d639a3c0209dab2e8797e91fb1d008e7838ffb5c/intray-server/src/Intray/Server/Handler/Utils.hs#L21-L24

Reply:
Hi Syd. Changes requested. I implemented this change before looking at your references - I try to always do this since it is helping me learn rather then how to copy-and-paste. My solution and your reference are almost identical. That's either a testament to Haskell's type system or I am getting better at this. (Or both!)